plugins/org.eclipse.reddeer.workbench/src/org/eclipse/reddeer/workbench/api/Editor.java

```diff
@@ -13,90 +13,112 @@
+
+	/**
+	 * Closes this editor.
+	 * 
+	 * @param save
+	 *            If true, content will be saved
+	 */
+	void close(boolean save);
+
+	/**
+	 * Opens content assistant.
+	 * 
+	 * @return Content assistant shell
+	 */
+	ContentAssistant openContentAssistant();
+
+	/**
+	 * Opens content assistant of specified type
+	 * 
+	 * @param assistantType
+	 *            type of content assistant to open
+	 * @return Content assistant shell
+	 */
+	ContentAssistant openContentAssistant(ContentAssistantEnum assistantType);
+
+	/**
+	 * Opens content assistant of specified type
+	 * 
+	 * @param assistantLabel
+	 *            type of content assistant to open
+	 * @return Content assistant shell
+	 */
+	ContentAssistant openContentAssistant(String assistantLabel);
+
+	/**
 	 * Retrieves callable content assistants.
+	 * 
 	 * @return List of assistants enums.
 	 */
-    List<ContentAssistantEnum> getAvailableContentAssistants();
-
-    /**
-     * Opens quickfix content assistant.
-     * @return Content assistant shell
-     */
-    ContentAssistant openQuickFixContentAssistant();
-
-    /**
-     * Opens open on assistant.
-     * @return Content assistant shell
-     */
-    ContentAssistant openOpenOnAssistant();
-
-    /**
-     * Returns editor validation markers.
-     * @return editor validation markers
-     */
-    List<Marker> getMarkers();
-    
-    /**
-     * Gets content assistant opened automatically by instructions defined within run method
-     * of execute parameter or null in case Content Assistant shell was not opened.
-     *
-     * @param execute the execute
-     * @return Content assistant
-     */
-    ContentAssistant getAutoContentAssistant(Runnable execute);
-
-    /**
-     * Returns an editor file associated to the editor.
-     * 
-     * @return Editor file associated to the editor
-     */
-    EditorFile getAssociatedFile();  
+	List<ContentAssistantEnum> getAvailableContentAssistants();
+
+	/**
+	 * Opens quickfix content assistant.
+	 * 
+	 * @return Content assistant shell
+	 */
+	ContentAssistant openQuickFixContentAssistant();
+
+	/**
+	 * Opens open on assistant.
+	 * 
+	 * @return Content assistant shell
+	 */
+	ContentAssistant openOpenOnAssistant();
+
+	/**
+	 * Returns editor validation markers.
+	 * 
+	 * @return editor validation markers
+	 */
+	List<Marker> getMarkers();
+
+	/**
+	 * Gets content assistant opened automatically by instructions defined within run method of execute parameter or
+	 * null in case Content Assistant shell was not opened.
+	 *
+	 * @param execute
+	 *            the execute
+	 * @return Content assistant
+	 */
+	ContentAssistant getAutoContentAssistant(Runnable execute);
+
+	/**
+	 * Returns an editor file associated to the editor.
+	 * 
+	 * @return Editor file associated to the editor
+	 */
+	EditorFile getAssociatedFile();
+
+	/**
+	 * Returns a context menu associated to the editor.
+	 * 
+	 * @return Context menu associated to the editor
+	 */
+	Menu getContextMenu();
```

Comment:
we should add this to WorkbenchPart (and make it work with views too)

Reply:
This can be implemented later without breaking existing API, see #1854

Reply:
@rawagner so this PR can be merged ;-)